packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -34,6 +34,16 @@ export default declare((api, options) => {
             optionalPath.node.type = "CallExpression";
             optionalPath = optionalPath.get("callee");
           }
+
+          // unwrap a TSNonNullExpression if need
+          if (optionalPath.isTSNonNullExpression()) {
+            optionalPath.replaceWith(optionalPath.node.expression);
+
+            if (optionalPath.isOptionalMemberExpression()) {
```

Comment:
I think we should only skip TSNonNullExpression and leave the `node.expression` as-is. Because the non-null expression does not imply any runtime checks, which is done by the optional member.

Reply:
Oh, yeah. I tested now removing this `if` block and works fine. Okay, I'm updating this PR. Thank you.